projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,12 +1,56 @@
-import { KeyValue } from '@angular/common';
+import { Dictionary } from '@hypertrace/common';
 import { SelectOption } from '../../select/select-option';
+import { TableFilter } from '../table-api';
 
-export interface SelectFilter {
+export interface SelectControl {
   placeholder?: string;
-  options: SelectOption<KeyValue<string, unknown>>[];
+  default?: SelectOption<TableControlOption>;
+  options: SelectOption<TableControlOption>[];
 }
 
 export interface SelectChange {
-  select: SelectFilter;
-  value: KeyValue<string, unknown>;
+  select: SelectControl;
+  value: TableControlOption;
 }
+
+export interface CheckboxControl {
+  label: string;
+  value: boolean;
+  options: TableCheckboxOptions;
+}
+
+export interface CheckboxChange {
+  checkbox: CheckboxControl;
+  value: boolean;
```

Comment:
isn't the value contained in the option being sent back (similar to select)?

Reply:
Oh, good eye. Returning `option` was a later change that supersedes returning `value`. Removing.